src/rpc/net.cpp

```diff
@@ -234,12 +234,14 @@ UniValue addnode(const JSONRPCRequest& request)
 
 UniValue disconnectnode(const JSONRPCRequest& request)
 {
-    if (request.fHelp || request.params.size() != 1)
+    if (request.fHelp || request.params.size() >= 3)
         throw std::runtime_error(
-            "disconnectnode \"address\" \n"
+            "disconnectnode \"address\" [nodeid]\n"
             "\nImmediately disconnects from the specified node.\n"
             "\nArguments:\n"
             "1. \"address\"     (string, required) The IP address/port of the node\n"
+            "2. \"nodeid\"   (string, required) The node ID(see getpeerinfo for nodes)\n"
```

Comment:
Please mention in the documentation that only one of either can be provided, and the other one needs to be null.
Also I'd prefer a space after `ID` before `(` otherwise it looks like a parametrized something